In [7]:
# ============================================================
# TEST — LIVE PIPELINE
# ============================================================

import sys
sys.path.append(r"C:\Users\tlili\OneDrive\Bureau\Bootcamp\AI-powered-cryptocurrency-market-analysis-and-decision-support-system\src")

from live_pipeline import predict, build_live_features

# ── Test 1 : build features only (no model needed)
print("="*55)
print("  TEST 1 — BUILD FEATURES (BTC)")
print("="*55)
features_btc = build_live_features("btc")
print(f"\n  Feature vector:")
for name, val in features_btc.items():
    print(f"    {name:<25} : {val:.6f}")

# ── Test 2 : build features ETH
print("\n" + "="*55)
print("  TEST 2 — BUILD FEATURES (ETH)")
print("="*55)
features_eth = build_live_features("eth")
print(f"\n  Feature vector:")
for name, val in features_eth.items():
    print(f"    {name:<25} : {val:.6f}")

# ── Test 3 : full prediction BTC
print("\n" + "="*55)
print("  TEST 3 — FULL PREDICTION (BTC)")
print("="*55)
result_btc = predict("btc")

# ── Test 4 : full prediction ETH
print("\n" + "="*55)
print("  TEST 4 — FULL PREDICTION (ETH)")
print("="*55)
result_eth = predict("eth")

# ── Summary
print("\n" + "="*55)
print("  SUMMARY")
print("="*55)
print(f"  BTC → {result_btc['direction']} | Confidence: {result_btc['confidence']}%")
print(f"  ETH → {result_eth['direction']} | Confidence: {result_eth['confidence']}%")

  TEST 1 — BUILD FEATURES (BTC)

  LIVE PIPELINE — BTC
  Warmup window : 2025-10-18 → 2026-02-15

📥 Fetching crypto prices...
  ✅ BTC-USD: 119 rows

📥 Fetching macro signals...
  ✅ SPY: 81 rows | 2025-10-20 → 2026-02-13
  ✅ DX-Y.NYB: 81 rows | 2025-10-20 → 2026-02-13
  ✅ ^VIX: 81 rows | 2025-10-20 → 2026-02-13

📥 Fetching Fear & Greed...
  ✅ Fear & Greed: 150 rows | 2025-09-19 → 2026-02-15

⚙️  Computing macro returns...

⚙️  Merging signals...
  Shape after merge: (119, 9)

⚙️  Preprocessing...

⚙️  Computing features...
  └─ Feature engineering complete ✅

  Date        : 2026-02-13
  Price       : $68,857.84
  Features    : 21 ✅
  NaN check   : 0 (should be 0)

  Feature vector:
    price_to_ma7              : 0.003063
    price_to_ma30             : -0.152232
    price_to_ma50             : -0.189622
    rsi_14                    : 31.664848
    macd_histogram            : -408.859916
    momentum_acceleration     : 0.069563
    volatility_7d             : 0.023009
    volatility_2

c:\Users\tlili\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2739: UserWarning: X does not have valid feature names, but RandomForestClassifier was fitted with feature names
  warnings.warn(
c:\Users\tlili\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2739: UserWarning: X does not have valid feature names, but GradientBoostingClassifier was fitted with feature names
  warnings.warn(
c:\Users\tlili\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2739: UserWarning: X does not have valid feature names, but RandomForestClassifier was fitted with feature names
  warnings.warn(
c:\Users\tlili\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2739: UserWarning: X does not have valid feature names, but GradientBoostingClassifier was fitted with feature names
  warnings.warn(


  ✅ SPY: 81 rows | 2025-10-20 → 2026-02-13
  ✅ DX-Y.NYB: 81 rows | 2025-10-20 → 2026-02-13
  ✅ ^VIX: 81 rows | 2025-10-20 → 2026-02-13

📥 Fetching Fear & Greed...
  ✅ Fear & Greed: 150 rows | 2025-09-19 → 2026-02-15

⚙️  Computing macro returns...

⚙️  Merging signals...
  Shape after merge: (119, 9)

⚙️  Preprocessing...

⚙️  Computing features...
  └─ Feature engineering complete ✅

  Date        : 2026-02-13
  Price       : $2,048.53
  Features    : 21 ✅
  NaN check   : 0 (should be 0)

  PREDICTION — ETH
  Date       : 2026-02-15
  Direction  : UP 📈
  Confidence : 60.39%
  Model      : GradientBoosting (tuned)

  SUMMARY
  BTC → UP 📈 | Confidence: 70.54%
  ETH → UP 📈 | Confidence: 60.39%


c:\Users\tlili\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2739: UserWarning: X does not have valid feature names, but GradientBoostingClassifier was fitted with feature names
  warnings.warn(
c:\Users\tlili\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2739: UserWarning: X does not have valid feature names, but GradientBoostingClassifier was fitted with feature names
  warnings.warn(


### Example to interpret the predection :

Example:
  P(DOWN) = 0.2946  →  29.46%
  P(UP)   = 0.7054  →  70.54%  ← model picks this (higher)
  
  Direction  = UP
  Confidence = 70.54%

  | Confidence | Meaning |
|------------|----------|
| 50% | Model has no idea  equivalent to a coin flip |
| 55–60% | Weak signal  low conviction |
| 60–70% | Moderate signal  some predictive edge |
| 70–80% | Strong signal act with reasonable conviction |
| 80%+ | Very strong signal rare, high conviction prediction |


BTC at 70.54% means:

- The model is fairly confident BTC will go UP tomorrow  not a coin flip, a real signal with meaningful edge.

ETH at 60.39% means:

- The model leans UP but with lower conviction  the signal is weaker, more uncertainty.